components/match2/wikis/naraka/match_group_input_custom.lua

```diff
@@ -0,0 +1,188 @@
+		local winnerInput = map.winner --[[@as string?]]
+
+		Table.mergeInto(map, MatchGroupInputUtil.readDate(map.date))
+		map.finished = MatchGroupInputUtil.mapIsFinished(map)
+
+		map.opponents = Array.map(opponents, function(matchOpponent)
+			local opponentMapInput = Json.parseIfString(matchOpponent['m' .. mapIndex])
+			return MapFunctions.makeMapOpponentDetails(opponentMapInput, scoreSettings)
+		end)
+
+		map.scores = Array.map(map.opponents, Operator.property('score'))
+		if map.finished then
+			map.status = MatchGroupInputUtil.getMatchStatus(winnerInput, finishedInput)
+			map.winner = MatchGroupInputUtil.getWinner(map.status, winnerInput, map.opponents)
+		end
+
+		map.extradata = MapFunctions.getExtraData(map)
+
+		table.insert(maps, map)
+		match[key] = nil
+	end
+
+	return maps
+end
+
+---@param opponents table[]
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(opponents, maps)
+	return function(opponentIndex)
+		return Array.reduce(Array.map(maps, function(map)
+			return map.opponents[opponentIndex].score or 0
+		end), Operator.add, 0) + (opponents[opponentIndex].extradata.startingpoints or 0)
+	end
+end
+
+
+---@param match table
+---@return {score: table, status: table}
+function MatchFunctions.parseSettings(match, opponentCount)
+	-- Score Settings
+	local scoreSettings = {
+		kill = Array.map(Array.range(1, opponentCount), function(index)
+			return tonumber(match['p' .. index .. '_kill']) or tonumber(match.p_kill) or 1
+		end),
+		placement = Array.map(Array.range(1, opponentCount), function(index)
+			return tonumber(match['p' .. index]) or 0
+		end)
+	}
+
+	-- Status colors (up/down etc)
+	local statusParsed = {}
+	Array.forEach(Array.parseCommaSeparatedString(match.bg, ','), function (status)
+		local placements, color = unpack(Array.parseCommaSeparatedString(status, '='))
+		local pStart, pEnd = unpack(Array.parseCommaSeparatedString(placements, '-'))
+		local pStartNumber = tonumber(pStart) --[[@as integer]]
+		local pEndNumber = tonumber(pEnd) or pStartNumber
+		Array.forEach(Array.range(pStartNumber, pEndNumber), function(placement)
+			statusParsed[placement] = color
+		end)
+	end)
+
+	local statusSettings = Array.map(Array.range(1, opponentCount), function(index)
+		return statusParsed[index] or ''
+	end)
+
+	return {
+		score = scoreSettings,
+		status = statusSettings,
+		settings = {
+			showGameDetails = Logic.nilOr(Logic.readBoolOrNil(match.showgamedetails), true),
+		}
+	}
+end
+
+---@param match table
+---@param games table[]
+---@param opponents table[]
+---@param settings table
+---@return table
+function MatchFunctions.getExtraData(match, games, opponents, settings)
+	return {
+		scoring = settings.score,
+		status = settings.status,
+		settings = settings.settings,
+	}
+end
+
+--
+-- map related functions
+--
+
+---@param map table
+---@return table
+function MapFunctions.getExtraData(map)
+	return {
+		dateexact = map.dateexact,
```

Comment:
should imo standardize the stored value in stead of just storing the plain input
```suggestion
		dateexact = tostring(Logic.readBool(map.dateexact)),
```
or
```suggestion
		dateexact = Logic.readBool(map.dateexact) and 1 or 0,
```

Reply:
It's not raw input, it comes from readDate